### DISCUSSION TITLE
One bundle per page

### DISCUSSION THREAD

**Discussion by brillout** [vikejs/vike#642] [Discussion] [answered]
> I am trying different chunk strategies, using manualChunk. I wanna see if maybe it would make sense to chunk per page.
> ```js
> manualChunks: (id) => {
>   if (id.includes('page.client.tsx')) {
>     return id
>       .split('/')
>       .pop()
>       ?.replace(/[^a-z]/g, '_');
>   }
> 
>   return undefined;
> },
> ```
> To my surprise, this produces entry-entry-client-routing.82f3a8dc.js that imports other pages.
> 
> What's the reason other pages are loaded here?
> 
> https://gist.github.com/gajus/fef9c7cd1c4bd6b7581321faf7314a08

**Comment by brillout** [comment, marked as answer]
You'll be able to achieve that with the V1 design I'm currently working on (because, with V1, each page gets a unique client entry). Until the V1 design is released (ETA 1-2 weeks), I don't think there is an easy way to achieve that.

**Comment by abecks** [comment]
@brillout Now that v1 has released do you have any plans for this? I just discovered Vike, and I'm loving it so far, but have a similar use case where I want one of the pages to generate a single JS file instead of chunks. The rest of the pages should split normally.

**Reply by brillout** [reply]
The chunking is done by Rollup and I don't know whether Rollup supports that use case, see Rollup's chunking options.
